include/Operator/OneElectronOperator.hpp

```diff
@@ -61,35 +60,27 @@ class OneElectronOperator : public SquareMatrix<Scalar>, public Operator<Scalar>
 
 
     /*
-     *  OVERRIDDEN PUBLIC METHODS
+     *  PUBLIC METHODS
      */
 
     /**
      *  In-place transform the matrix representation of the one-electron operator
      *
+     *  @tparam TransformationScalar        the type of scalar used for the transformation matrix
+     *
      *  @param T    the transformation matrix between the old and the new orbital basis, it is used as
      *      b' = b T ,
      *   in which the basis functions are collected as elements of a row vector b
+     *
+     *  Note that in order to use these transformation formulas, the multiplication between TransformationScalar and Scalar should be 'enabled'. See LinearCombination.hpp for an example
      */
-    void transform(const SquareMatrix<Scalar>& T) override {
+    template <typename TransformationScalar = Scalar>
```

Comment:
Yes, that is (almost) correct. Let me explain why I chose this approach.

Without a `TransformationScalar`, we would only be able to transform `OneElectronOperator<double>` with a `SquareMatrix<double>`. Obviously, this wouldn't be as flexible as we would like, because we would like to have the following transformations:
- transform `OneElectronOperator<double>` with a complex transformation matrix (and the other way around)
- transform `OneElectronOperator<CustomScalar>` with a transformation matrix that doesn't consist of a `CustomScalar`. For example, this is the case that covers `OneElectronOperator<LC<GTO>>.transform(SquareMatrix<double>)`.

If then `CustomScalar` is made an `Eigen` scalar type (such as we did in `LinearCombination.hpp` and we let `ScalarBinaryOpTraits` know of the (scalar) multiplication between `CustomScalar` and `TransformationScalar`, we are good to go.